odl/solvers/util/callback.py

```diff
@@ -854,6 +855,113 @@ def __repr__(self):
             self.logy)
 
 
+class CallbackPrintHardwareUsage(SolverCallback):
+
+    """Callback for printing memory and CPU usage.
+
+    To use this callback, the user needs to have the package ``psutil``
+    installed.
+    """
+
+    def __init__(self, step=1, print_cpu=True, print_mem=True,
+                 print_swap=True, fmt_cpu='CPU usage (% each core): {}',
+                 fmt_mem='RAM usage: {}', fmt_swap='SWAP usage: {}'):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        step : positive int, optional
+            Number of iterations between output. Default: 1
+        print_cpu : bool, optional
```

Comment:
But you can keep the old names for the format string parameters, they're good. Add a sentence in the format string docs then that "An empty format string disables printing of XYZ usage".